### PR TITLE
[Dynamic Selection] Addressing comments for dynamic_selection.h and auto_tune

### PR DESCRIPTION
**Pull request by AnuyaWelling2801**
1. std::decay<T>::type to std::decay_t<T>
2. Fixed definition of submit to not use policy's function directly
3. Checking use of forwards in submit and submit_and_wait
4. move internal namespace classes into auto_tune_policy as private types
5. change auto_tune to use steady_clock

### CODE REVIEW

**Review by vossmjp — CHANGES_REQUESTED** [review]
Missing check for backend.

File: include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h [hunk 1]
```diff
@@ -23,53 +23,20 @@
 namespace oneapi {
 namespace dpl {
 namespace experimental {
-#if _DS_BACKEND_SYCL != 0
   template <typename Backend=sycl_backend, typename... KeyArgs>
```
[the remark this answers is not in the document]
Comment (AnuyaWelling2801) [reply include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h:26]: Fixed

**Review by vossmjp — APPROVED** [review]
Still need to check for backend

File: include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h [hunk 2]
```diff
@@ -23,53 +23,20 @@
 namespace oneapi {
 namespace dpl {
 namespace experimental {
-#if _DS_BACKEND_SYCL != 0
```
Comment (egfefey) [inline include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h:26]: Why was this guard removed. Non sycl tests will hit this and fail.

**Pull request merged**